go/azidentity/azure_identity_credential_adapter.go

```diff
@@ -0,0 +1,23 @@
+package azidentity
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// NewAzureIdentityCredentialAapter returns BearerAuthorizer.
+// azureCredential: TokenCredential
+// scopes: The list of scopes for which the token will have access
+func NewAzureIdentityCredentialAapter(tokenCredential azcore.TokenCredential, scopes []string) (*autorest.BearerAuthorizer, error) {
+	var err error
+	if tokenCredential == nil {
```

Comment:
I don't think the base adapter should default to any kind of credential.  We can build higher-level constructs on top of this that do.

Reply:
@jhendrixMSFT - Just FYI that all the other langs default to DAC.  Given that, do you still feel like Go shouldn't default to DAC?